extract/raw_array_cohort_extract.py

```diff
@@ -100,14 +100,14 @@ def get_subselect(fq_array_table, samples, id, extract_genotype_counts_only):
         subs[id] = get_subselect(fq_array_table, samples, id, extract_genotype_counts_only)
         j = j + 1
 
-  # ref vs alt allele doesn't matter for HWE or call rate
   # TODO: make genotype flexible to older encodings
+  # TODO: push this logic into the GATK tool that reads the table so that the intermediate genotype counts table has true counts for 1/2 and 2/2 genotypes
   select_sql = (
                 f" (SELECT probe_id, " +
-                f"COUNT(IF(GT_encoded LIKE 'AA', Sample_id, null)) hom_ref, \n" +
-                f"COUNT(IF(GT_encoded LIKE 'AB', Sample_id, null)) het, \n" +
-                f"COUNT(IF(GT_encoded LIKE 'BB', Sample_id, null)) hom_var, \n" +
-                f"COUNT(IF(GT_encoded LIKE '.', Sample_id, null)) no_call \n" +
+                f"SUM(IF(GT_encoded is null OR GT_encoded = 'R', 1, 0)) hom_ref, \n" +
+                f"SUM(IF(GT_encoded = 'X', 1, 0)) het, \n" +
+                f"SUM(IF(GT_encoded = 'A' OR GT_encoded = 'Y' OR GT_encoded = 'B', 1, 0)) hom_var, \n" +
```

Comment:
should this be 'AA' and 'BB'?

Reply:
actually -- what is this? :D

Reply:
This is the new genotype encoding, but I'm putting all 1/2 and 2/2 genotypes (encoded as Y and B) into the 1/1 count. If we think we'll want the full counts for all genotypes I can separate it out. The HWE code in GATK can't handle more than two alleles, but I can put however we want to handle that into the GATK tool instead of here. Does that sound better than leaving the logic in this query?